Solved issue 3392.

DustyDogCodex opened the pull request:
Solves issue #3392 

 Added an exp.md with an introduction to the math.exp() method with syntax and examples!

<!--
👋 Hi, thanks for sending a PR to Codecademy Docs! 💖
Please fill out all fields below to ensure your PR is reviewed quickly.

**IMPORTANT**

If you would like to receive credit for your contribution to an entry, make sure to have your Codecademy user profile linked to your GitHub account:

1. Go to your Codecademy dashboard.
2. Click on your profile image in the top-right and then choose "Profile".
3. Then click "Edit Profile".
4. In the "GitHub Username" field, add your username (without the @).
5. Click "Save Changes"

Of course, you can opt not to do this and be listed as an "Anonymous contributor", instead. :)
-->

### Description

Created a new file named exp.md at docs/content/lua/concepts/mathematical-library/terms/exp/exp.md. Added a brief description of the math.exp() function in Lua, alongside syntax and examples.

### Type of Change

<!--- Please delete or cross off the bullet point(s) that are irrelevant to this PR: -->

- Adding a new entry

### Checklist

<!-- Please check ALL the boxes: -->

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

<!--
Having trouble with the PR checker? Here are some common issues and resolutions:

- verify_formatting is failing
  - run `yarn format path/to/markdown/file.md` or `yarn format:all` and commit the results
- verify_lint is failing
  - same as above
  - if verify_lint is still failing, running `yarn lint` locally should let you know what needs to be changed by hand
- test is failing
  - ensure any new markdown files have a `Title` and `Description` defined in their metadata
  - ensure any new markdown files only contain alphanumerics and dashes in their file names and have the same name as their parent directory
  - if that looks ok, running `yarn test` locally should let you know what the issue is
-->

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/Codecademy/docs?pullRequest=3430) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/Codecademy/docs?pullRequest=3430) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/Codecademy/docs?pullRequest=3430) it.</sub>

Jamesllllllllll commented:
Hey @DustyDogCodex ! 

I'm a fellow contributor (not a maintainer) just dropping by to let ya know that you can edit and start typing `#3392`  into your description above, and select the issue that is related to this pull request from a drop-down menu.

It will then "link" this PR to that pull request!

Thanks for contributing <3

DustyDogCodex commented:
Hi @Jamesllllllllll !

Thanks for reminding of that! I completely forgot Github has that feature haha I appreciate your help!

SSwiniarski commented on content/lua/concepts/mathematical-library/terms/exp/exp.md:
```diff
@@ -0,0 +1,44 @@
+---
+Title: 'Exp'
+Description: 'Exp is method used to calculate the exponential value of a number.'
```
```suggestion
Description: 'Used to calculate the exponential value of a number.'
```

SSwiniarski commented on content/lua/concepts/mathematical-library/terms/exp/exp.md:
```diff
@@ -0,0 +1,44 @@
+  - 'paths/computer-science'
+---
+
+In Lua, the **math.exp()** function is a standard function that is part of the Lua 'math' library. The math.exp() function is used to calculate the exponential value of a number, i.e, the exponential constant 'e' (which is roughly equal to 2.718281828459045...) raised to a given value. Exponents are a very common part of mathematical operations across a wide variety of fields and disciplines such ...
```
```suggestion
In Lua, the **`math.exp()`** function is a standard function that is part of the Lua `math` library. The `math.exp()` function is used to calculate the exponential value of a number, i.e, the exponential constant 'e' (which is roughly equal to 2.718281828459045...) raised to a given value. Exponents are a very common part of mathematical operations across a wide variety of fields and disciplines such as physics, engineering, probability and statistics, and even machine learning! This means that the `math.exp()` function is a very useful and widely applicable tool for Lua developers.
```

SSwiniarski commented on content/lua/concepts/mathematical-library/terms/exp/exp.md:
```diff
@@ -0,0 +1,44 @@
+
+In Lua, the **math.exp()** function is a standard function that is part of the Lua 'math' library. The math.exp() function is used to calculate the exponential value of a number, i.e, the exponential constant 'e' (which is roughly equal to 2.718281828459045...) raised to a given value. Exponents are a very common part of mathematical operations across a wide variety of fields and disciplines such ...
+
+As a side-note, the **math.log()** function in the math library returns the inverse of the math.exp() function.
```
```suggestion
As a side-note, the **`math.log()`** function in the math library returns the inverse of the `math.exp()` function.
```

SSwiniarski commented on content/lua/concepts/mathematical-library/terms/exp/exp.md:
````diff
@@ -0,0 +1,44 @@
+
+As .exp() is a method that is a part of the standard Lua Math library, to use this function it must be called as 'math.exp()'
+
+```lua
+
+math.exp(mycoolvalue) -- returns the result of e raised to the power of mycoolvalue
+
+```
````
````suggestion
```pseudo
math.exp(mycoolvalue) -- returns the result of e raised to the power of mycoolvalue
```
````

SSwiniarski commented on content/lua/concepts/mathematical-library/terms/exp/exp.md:
```diff
@@ -0,0 +1,44 @@
+
+## Syntax
+
+As .exp() is a method that is a part of the standard Lua Math library, to use this function it must be called as 'math.exp()'
```
```suggestion
As `.exp()` is a method that is a part of the standard Lua Math library, to use this function it must be called as `math.exp()`
```

SSwiniarski commented on content/lua/concepts/mathematical-library/terms/exp/exp.md:
````diff
@@ -0,0 +1,44 @@
+
+```
+
+## Example
````
There are some things that need to be changed with this example. The code needs to be changed so the code produces some output to the console. Then, instead of a comment in the code, the output needs to be shown in a ` ```shell` block after the example like this:

````
This example results in the following output:

```shell
Output from example
```
````
We also need an introductory sentence before the example summarizing what the example is doing.

SSwiniarski reviewed (changes requested):
@DustyDogCodex I did a first review and left a couple of comments.

DustyDogCodex commented:
Hi @SSwiniarski !

Thank you for your code review! I appreciate you taking the time to help improve this PR :)

I have implemented the changes you suggested, and I have changed the structure of the examples to fit what you recommended. Additionally, I split the examples section into 2 sections (Example 1 and Example 2) for better readability. 

I have pushed a commit with the new changes. Please let me know if there is anything else that needs to be updated.

Thanks!

SSwiniarski commented on content/lua/concepts/mathematical-library/terms/exp/exp.md:
```diff
@@ -0,0 +1,72 @@
+
+## Syntax
+
+As `.exp()` is a method that is a part of the standard Lua Math library, to use this function it must be called as `math.exp()`
```
```suggestion
As `.exp()` is a method that is a part of the standard Lua `math` library, to use this function it must be called as `math.exp()`
```

SSwiniarski reviewed (changes requested):
@DustyDogCodex One more comment.

DustyDogCodex commented:
hi @SSwiniarski !

Hope you are doing well! I have implemented the changes you recommended in your code review and this PR is ready for another review.

Thanks!

SSwiniarski reviewed (approved):
@DustyDogCodex Looks good for a second review.

caupolicandiaz commented on content/lua/concepts/mathematical-library/terms/exp/exp.md:
```diff
@@ -0,0 +1,72 @@
+Description: 'Used to calculate the exponential value of a number.'
+Subjects:
+  - 'Computer Science'
+  - 'Lua'
```
```suggestion
```

caupolicandiaz commented on content/lua/concepts/mathematical-library/terms/exp/exp.md:
```diff
@@ -0,0 +1,72 @@
+  - 'paths/computer-science'
+---
+
+In Lua, the **`math.exp()`** function is a standard function that is part of the Lua `math` library. The `math.exp()` function is used to calculate the exponential value of a number, i.e, the exponential constant 'e' (which is roughly equal to 2.718281828459045...) raised to a given value. Exponents are a very common part of mathematical operations across a wide variety of fields and disciplines  ...
```
```suggestion
In Lua, the **`math.exp()`** function is a standard function that is used to calculate the exponential value of a number, i.e, the exponential constant *'e'* (which is roughly equal to 2.718281828459045) raised to a given value. 

Exponents are a very common part of mathematical operations across a wide variety of fields and disciplines such as physics, engineering, probability and statistics, and even machine learning! This means that the `math.exp()` function is a very useful and widely applicable tool for Lua developers.
```

caupolicandiaz commented on content/lua/concepts/mathematical-library/terms/exp/exp.md:
```diff
@@ -0,0 +1,72 @@
+
+## Syntax
+
+As `.exp()` is a method that is a part of the standard Lua `math` library, to use this function it must be called as `math.exp()`
```
```suggestion
As `.exp()` is a method that is a part of the standard Lua `math` library, to use this function it must be called as `math.exp()`.
```

caupolicandiaz commented on content/lua/concepts/mathematical-library/terms/exp/exp.md:
````diff
@@ -0,0 +1,72 @@
+As `.exp()` is a method that is a part of the standard Lua `math` library, to use this function it must be called as `math.exp()`
+
+```pseudo
+math.exp(mycoolvalue) -- returns the result of e raised to the power of mycoolvalue
````
```suggestion
math.exp(mycoolvalue) -- Returns the result of e raised to the power of mycoolvalue
```

caupolicandiaz commented on content/lua/concepts/mathematical-library/terms/exp/exp.md:
````diff
@@ -0,0 +1,72 @@
+1.0
+```
+
+This is because e raised to the power 0 (e^0) equals 1.
````
```suggestion
```

caupolicandiaz commented on content/lua/concepts/mathematical-library/terms/exp/exp.md:
```diff
@@ -0,0 +1,72 @@
+
+## Example 1
+
+To find the exponential value of 0,
```
```suggestion
To find the exponential value of `0`,
```

caupolicandiaz commented on content/lua/concepts/mathematical-library/terms/exp/exp.md:
```diff
@@ -0,0 +1,72 @@
+
+## Example 2
+
+Similarly, to find the exponential value of 8,
```
```suggestion
Similarly, to find the exponential value of `8`,
```

caupolicandiaz commented on content/lua/concepts/mathematical-library/terms/exp/exp.md:
````diff
@@ -0,0 +1,72 @@
+2980.9579870417
+```
+
+This equals the value of e raised to the power 8 (e^8).
````
```suggestion
```

caupolicandiaz reviewed (changes requested):
@DustyDogCodex I've added my comments. Please update, and tag me when you've made the changes. Thanks

caupolicandiaz reviewed (approved):
@DustyDogCodex @SSwiniarski LGTM
